packages/mongo/oplog_tests.js

```diff
@@ -180,6 +182,116 @@ process.env.MONGO_OPLOG_URL && Tinytest.addAsync(
+  const ExcludeCollection = new Mongo.Collection(excludeCollectionName);
+
+  const shouldBeTracked = new Promise((resolve) => {
+    IncludeCollection.find({ include: 'yes' }).observeChanges({
+      added(id, fields) { resolve(true) }
+    });
+  });
+  const shouldBeIgnored = new Promise((resolve, reject) => {
+    ExcludeCollection.find({ include: 'no' }).observeChanges({
+      added(id, fields) { 
+        // should NOT fire, because this is an excluded collection:
+        reject(false);
+      }
+    });
+    // we give it just 2 seconds until we resolve this promise:
+    setTimeout(() => {
+      resolve(true);
+    }, 2000);
+  });
+
+  // do the inserts:
+  await IncludeCollection.rawCollection().insertOne({ include: 'yes', foo: 'bar' });
+  await ExcludeCollection.rawCollection().insertOne({ include: 'no', foo: 'bar' });
+
+  test.equal(await shouldBeTracked, true);
+  test.equal(await shouldBeIgnored, true);
+
+  // reset the package settings:
+  delete Meteor.settings.packages.mongo;
+}
+
+process.env.MONGO_OPLOG_URL && Tinytest.addAsync(
+  'mongo-livedata - oplog - oplogSettings - oplogExcludeCollections',
+  async test => {
+    const collectionNameA = "oplog-a-" + Random.id();
+    const collectionNameB = "oplog-b-" + Random.id();
+    const mongoPackageSettings = {
+      oplogExcludeCollections: [collectionNameB]
+    };
+    await oplogOptionsTest({
+      test,
+      includeCollectionName: collectionNameA,
+      excludeCollectionName: collectionNameB,
+      mongoPackageSettings
+    });
+  }
+);
+
+process.env.MONGO_OPLOG_URL && Tinytest.addAsync(
+  'mongo-livedata - oplog - oplogSettings - oplogIncludeCollections',
+  async test => {
+    const collectionNameA = "oplog-a-" + Random.id();
+    const collectionNameB = "oplog-b-" + Random.id();
+    const mongoPackageSettings = {
+      oplogIncludeCollections: [collectionNameB]
+    };
+    await oplogOptionsTest({
+      test,
+      includeCollectionName: collectionNameB,
+      excludeCollectionName: collectionNameA,
+      mongoPackageSettings
+    });
+  }
+);
+
+process.env.MONGO_OPLOG_URL && Tinytest.addAsync(
+  'mongo-livedata - oplog - oplogSettings - oplogExcludeCollections & oplogIncludeCollections',
+  async test => {
+    // should fail, because we don't allow including and excluding at the same time!
+    const collectionNameA = "oplog-a-" + Random.id();
+    const collectionNameB = "oplog-b-" + Random.id();
+    const mongoPackageSettings = {
+      oplogIncludeCollections: [collectionNameA],
+      oplogExcludeCollections: [collectionNameB]
+    };
+    try {
+      await oplogOptionsTest({
+        test,
+        includeCollectionName: collectionNameA,
+        oplogExcludeCollections: collectionNameB,
+        mongoPackageSettings
+      });
+      // reset the package settings:
+      delete Meteor.settings.packages.mongo;
+      MongoInternals.defaultRemoteCollectionDriver().mongo._setOplogHandle(defaultOplogHandle);
```

Comment:
As I wrote above - this should be handled in `oplogOptionsTest`.

Reply:
different in this case, because for this test `oplogOptionsTest()` actually throws an error because of using both options - like it should! That's why here the teardown actually happens outside of `oplogOptionsTest()`!

Reply:
That only suggests that the cleanup should happen in a `finally` block in `oplogOptionsTest` - then it'll happen even if the test fails.